File: apps/backend/lib/store.ts
export interface AbstractStore {
  saveStartup(startupId: string, data: Record<string, unknown>): Promise<void>
  getStartup(startupId: string): Promise<Record<string, unknown> | null>
  getAllStartups(): Promise<Record<string, unknown>[]>

  saveLinkage(linkage: Record<string, unknown>): Promise<void>
  getLinkage(linkageId: string): Promise<Record<string, unknown> | null>
  getAllLinkages(): Promise<Record<string, unknown>[]>
  updateLinkage(linkageId: string, updates: Record<string, unknown>): Promise<Record<string, unknown> | null>

  savePartner(partnerId: string, data: Record<string, unknown>): Promise<void>
}

File: apps/backend_py/services/store.py
from __future__ import annotations

from abc import ABC, abstractmethod


class AbstractStore(ABC):

    @abstractmethod
    async def save_startup(self, startup_id: str, data: dict) -> None: ...

    @abstractmethod
    async def get_startup(self, startup_id: str) -> dict | None: ...

    @abstractmethod
    async def get_all_startups(self) -> list[dict]: ...

    @abstractmethod
    async def save_linkage(self, linkage: dict) -> None: ...

    @abstractmethod
    async def get_linkage(self, linkage_id: str) -> dict | None: ...

    @abstractmethod
    async def get_all_linkages(self) -> list[dict]: ...

    @abstractmethod
    async def update_linkage(self, linkage_id: str, updates: dict) -> dict | None: ...

    @abstractmethod
    async def save_partner(self, partner_id: str, data: dict) -> None: ...
